controllers/topology/reconcile_state.go

```diff
@@ -269,15 +269,15 @@ func (r *ClusterReconciler) reconcileMachineDeployments(ctx context.Context, s *
 func (r *ClusterReconciler) createMachineDeployment(ctx context.Context, md *scope.MachineDeploymentState) error {
 	log := tlog.LoggerFrom(ctx).WithMachineDeployment(md.Object)
 
-	ctx, _ = log.WithObject(md.InfrastructureMachineTemplate).Into(ctx)
-	if err := r.reconcileReferencedTemplate(ctx, reconcileReferencedTemplateInput{
+	infraCtx, _ := log.WithObject(md.InfrastructureMachineTemplate).Into(ctx)
```

Comment:
I'm fine with keeping the old version, but it just seemed not ideal to me to stack loggers in the context on the "same level" (I realize it's a regular pattern when traversing deeper into a call stack).

With the previous code the context would have:

parent loggers
a logger for infra machine
a logger for bootstrap
It's not really a problem as the last one is used.